e2e/tests/block/create_block_test.go

```diff
@@ -0,0 +1,140 @@
+package block
+
+import (
+	"github.com/rook/rook/e2e/framework/clients"
+	"github.com/rook/rook/e2e/framework/contracts"
+	"github.com/rook/rook/e2e/tests"
+	"github.com/rook/rook/pkg/model"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"strings"
+	"testing"
+)
+
+var (
+	default_pool   = "rbd"
+	pool1          = "rook_test_pool"
+	blockImageName = "testImage"
+)
+
+func TestBlockCreate(t *testing.T) {
+	suite.Run(t, new(BlockImageCreateSuite))
+}
+
+type BlockImageCreateSuite struct {
+	suite.Suite
+	testClient      *clients.TestClient
+	rc              contracts.RestAPIOperator
+	init_blockCount int
+}
+
+func (s *BlockImageCreateSuite) SetupSuite() {
+
+	var err error
+
+	s.testClient, err = clients.CreateTestClient(tests.Platform)
+	require.Nil(s.T(), err)
+
+	s.rc = s.testClient.GetRestAPIClient()
+	initialBlocks, err := s.rc.GetBlockImages()
+	require.Nil(s.T(), err)
+	s.init_blockCount = len(initialBlocks)
+}
+
+//Test Creating Block image on default pool(rbd)
+func (s *BlockImageCreateSuite) TestCreatingNewBlockImageOnDefaultPool() {
+
+	s.T().Log("Test Creating new block image  for default pool")
+	newImage := model.BlockImage{Name: blockImageName, Size: 123, PoolName: default_pool}
+	cbi, err := s.rc.CreateBlockImage(newImage)
+	require.Nil(s.T(), err)
+	require.Contains(s.T(), cbi, "succeeded created image")
+	b, _ := s.rc.GetBlockImages()
+	require.Equal(s.T(), s.init_blockCount+1, len(b), "Make sure new block image is created")
+	s.T().Log("Test Passed")
+
+}
+
+//Test Creating Block image on custom pool
+func (s *BlockImageCreateSuite) TestCreatingNewBlockImageOnCustomPool() {
+
+	s.T().Log("Test Creating new block image for custom pool")
+	newPool := model.Pool{Name: pool1}
+	_, err := s.rc.CreatePool(newPool)
+	require.Nil(s.T(), err)
+
+	newImage := model.BlockImage{Name: blockImageName, Size: 123, PoolName: newPool.Name}
+	cbi, err := s.rc.CreateBlockImage(newImage)
+	require.Nil(s.T(), err)
+	require.Contains(s.T(), cbi, "succeeded created image")
+	b, _ := s.rc.GetBlockImages()
+	require.Equal(s.T(), s.init_blockCount+1, len(b), "Make sure new block image is created")
+	s.T().Log("Test Passed")
+
+}
+
+//Test Creating Block image twice on same pool
+func (s *BlockImageCreateSuite) TestRecreatingBlockImageForSamePool() {
+
+	s.T().Log("Test Case when Block Image is created with Name that is already used by another block on same pool")
```

Comment:
if the tests are run with verbose, Go will print the name of the function. Seems like that would be sufficient and this could just turn into a comment